glustercli/cmd/bitrot.go

```diff
@@ -0,0 +1,203 @@
+package cmd
+
+import (
+	"fmt"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+const (
+	helpBitrotCmd               = "Gluster Bitrot"
+	helpBitrotEnableCmd         = "Enable Bitrot"
+	helpBitrotDisableCmd        = "Disable Bitrot"
+	helpBitrotScrubThrottleCmd  = "Configure Scrub Throttle"
+	helpBitrotScrubFrequencyCmd = "Configure Scrub Frequency"
+	helpBitrotScrubCmd          = "Bitrot Scrub Command"
+)
+
+const (
+	scrubPause    = "pause"
+	scrubResume   = "resume"
+	scrubOndemand = "ondemand"
+)
+
+func init() {
+	// Bitrot Enable
+	bitrotCmd.AddCommand(bitrotEnableCmd)
+
+	// Bitrot Disable
+	bitrotCmd.AddCommand(bitrotDisableCmd)
+
+	// Configure scrub throttle
+	bitrotCmd.AddCommand(bitrotScrubThrottleCmd)
+
+	// Configure scrub frequency
+	bitrotCmd.AddCommand(bitrotScrubFrequencyCmd)
+
+	// Bitrot scrub command
+	bitrotCmd.AddCommand(bitrotScrubCmd)
+
+	RootCmd.AddCommand(bitrotCmd)
+}
+
+var bitrotCmd = &cobra.Command{
+	Use:   "bitrot",
+	Short: helpBitrotCmd,
+}
+
+var bitrotEnableCmd = &cobra.Command{
+	Use:   "enable <volname>",
+	Short: helpBitrotEnableCmd,
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		volname := args[0]
+
+		err := client.BitrotEnable(volname)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"volume": volname,
+				"error":  err.Error(),
+			}).Error("failed to enable bitrot")
+			failure(fmt.Sprintf("Failed to enable bitrot for volume %s\n", volname), 1)
+		}
+		fmt.Printf("Bitrot enabled successfully for volume %s\n", volname)
+	},
+}
+
+var bitrotDisableCmd = &cobra.Command{
+	Use:   "disable <volname>",
+	Short: helpBitrotDisableCmd,
+	Args:  cobra.ExactArgs(1),
+	Run: func(cmd *cobra.Command, args []string) {
+		volname := args[0]
+
+		err := client.BitrotDisable(volname)
+		if err != nil {
+			log.WithFields(log.Fields{
+				"volume": volname,
+				"error":  err.Error(),
+			}).Error("failed to disable bitrot")
+			failure(fmt.Sprintf("Failed to disable bitrot for volume %s\n", volname), 1)
+		}
+		fmt.Printf("Bitrot disabled successfully for volume '%s'\n", volname)
+	},
+}
+
+func bitrotValidScrubThrottleValue(value string) bool {
```

Comment:
These validator functions not required in CLI since API will validate the values.  In future Volume Set API can return different error codes for Invalid Key, Invalid Value etc that can be used in CLI. For now, we can remove the check from here.

Reply:
Done